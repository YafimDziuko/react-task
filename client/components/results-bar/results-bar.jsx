import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './results-bar.scss';

class ResultsBar extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <div className="results-bar">
                <p className="results-count">{this.props.count} movies found</p>
                <div className="results-sort">
                    <p>Sort by</p>
                    <p className={this.props.user.sort === "releaseDate" ? "cl-red" : "cl-black"} onClick={this.props.switchSort.bind(this, "releaseDate")}>release date</p>
                    <p className={this.props.user.sort === "rating" ? "cl-red" : "cl-black"} onClick={this.props.switchSort.bind(this, "rating")}>rating</p>
                </div>
            </div>
        );
    }
};

ResultsBar.propTypes = {
    count: PropTypes.number
};
  
ResultsBar.defaultProps = {
    count: 0
};



const mapStateToProps = (state) => {
    return {
        user: state
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        switchSort: (sort) => {
            dispatch({
                type: 'SWITCH_SORT',
                sort: sort,
            })
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ResultsBar);