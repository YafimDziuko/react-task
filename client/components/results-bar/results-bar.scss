@import "../../mixins.scss";

.results-bar {
    display: flex;
    justify-content: space-between;
    padding: 10px 150px;
    background-color: $light-grey;
    font-weight: 700;
    font-size: 14px;

    .results-sort {
        display: flex;
        flex-direction: row;
        justify-content: space-between;

        p:nth-child(2) {
            margin: 0 15px;
        }

        p:nth-child(2):hover,
        p:nth-child(3):hover {
            cursor: pointer;
        }

        .active {
            color: $red;
        }
    }
}

@media screen and ( max-width: $medium ) {
    .results-bar {
        padding: 10px 60px;
        font-size: 12px;
    }
}

@media screen and ( max-width: $small ) {
    .results-bar {
        padding: 10px 20px;
        font-size: 10px;
    }
}