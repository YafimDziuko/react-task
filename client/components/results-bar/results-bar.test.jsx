import React from 'react';
import { ResultsBar } from './results-bar';

describe('<ResultsBar/>', function() {
  it('should render ResultsBar and match snapshot', function() {
      const wrapper = shallow(<ResultsBar />);

      expect(wrapper).toMatchSnapshot();
  });

  it('should render ResultsBar with props', function() {
    const wrapper = shallow(<ResultsBar count={135}/>);

    expect(wrapper.contains(<p className="results-count">135 movies found</p>)).toEqual(true);
    expect(wrapper.state().rating).toEqual("active");
  });

  it('should switch sort by clicking', () => {
    const wrapper = mount(<ResultsBar />);
        const instance = wrapper.instance();
        
        jest.spyOn(instance, 'switchSort');
        wrapper.update();
        const buttons = wrapper.find('p');

        buttons.at(2).simulate('click');
      
        expect(instance.switchSort).toHaveBeenCalled();

        buttons.at(3).simulate('click');
      
        expect(instance.switchSort).toHaveBeenCalled();
        buttons.at(3).simulate('click');
      
        expect(instance.switchSort).toHaveBeenCalled();
   });
});