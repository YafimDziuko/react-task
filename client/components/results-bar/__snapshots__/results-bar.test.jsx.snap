// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsBar/> should render ResultsBar and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBar
    count={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="results-count"
        >
          0
           movies found
        </p>,
        <div
          className="results-sort"
        >
          <p>
            Sort by
          </p>
          <p
            className="passive"
            onClick={[Function]}
          >
            release date
          </p>
          <p
            className="active"
            onClick={[Function]}
          >
            rating
          </p>
        </div>,
      ],
      "className": "results-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            0,
            " movies found",
          ],
          "className": "results-count",
        },
        "ref": null,
        "rendered": Array [
          0,
          " movies found",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Sort by
            </p>,
            <p
              className="passive"
              onClick={[Function]}
            >
              release date
            </p>,
            <p
              className="active"
              onClick={[Function]}
            >
              rating
            </p>,
          ],
          "className": "results-sort",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sort by",
            },
            "ref": null,
            "rendered": "Sort by",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "release date",
              "className": "passive",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "release date",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "rating",
              "className": "active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "rating",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="results-count"
          >
            0
             movies found
          </p>,
          <div
            className="results-sort"
          >
            <p>
              Sort by
            </p>
            <p
              className="passive"
              onClick={[Function]}
            >
              release date
            </p>
            <p
              className="active"
              onClick={[Function]}
            >
              rating
            </p>
          </div>,
        ],
        "className": "results-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              0,
              " movies found",
            ],
            "className": "results-count",
          },
          "ref": null,
          "rendered": Array [
            0,
            " movies found",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Sort by
              </p>,
              <p
                className="passive"
                onClick={[Function]}
              >
                release date
              </p>,
              <p
                className="active"
                onClick={[Function]}
              >
                rating
              </p>,
            ],
            "className": "results-sort",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "release date",
                "className": "passive",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "release date",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "rating",
                "className": "active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "rating",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
